src/vde_backend/compute_power.py

```diff
@@ -136,25 +137,39 @@ def get_consumption_production_df(raw_df, sensors_config):
     return cons_prod_df
 
 
-def get_home_consumption_production_df(home_rawdata, home_id, sensors_df):
+def get_home_consumption_production_df(home_raw_data, home_config):
     """
     compute power data from raw data (coming from cassandra 'raw' table) :
     P_cons = P_tot - P_prod
     P_net = P_prod + P_cons
-    """
 
-    first_sid = list(home_rawdata.keys())[0]
-    cons_prod_df = home_rawdata[first_sid][["sensor_id", "day", "ts"]].copy()
-    cons_prod_df = cons_prod_df.rename(columns={"sensor_id": "home_id"})
-    cons_prod_df["home_id"] = home_id  # replace 1st sensor_id by home_id
-    cons_prod_df["P_cons"] = 0
-    cons_prod_df["P_prod"] = 0
-    cons_prod_df["P_tot"] = 0
+    :param home_raw_data:   Dictionary of dataframes with the following format:
+                            {sensor_id1: df, sensor_id2: df, ...} where the df has
```

Comment:
If the DataFrame has a `sensor_id` column already, there is no need to put it into a dictionary. All the data can be stored in a single DataFrame and you iterate over `home_raw_data.groupby('sensor_id')`.

Reply:
I reused the existing code (so the weird dict but with better explanation lol). I will change it rn